client/web/src/repo/RepoContainer.tsx

```diff
@@ -327,6 +322,11 @@ export const RepoContainer: FC<RepoContainerProps> = props => {
         onDidUpdateExternalLinks: setExternalLinks,
     }
 
+    const repoRevisionContainerSplats = [
+        ...(rawRevision ? [repoNameURL + `@${rawRevision}/*`] : []), // must exactly match how the revision was encoded in the URL
+        repoNameURL + '/*',
+    ]
```

Comment:
Is there a reason why we don't use relative matchers here given that we're in RR6 context?

If I use the following, the route mapping works but something else fails (Haven't researched yet):

```suggestion
    const repoRevisionContainerSplats = [
        ...(rawRevision ? [`@${rawRevision}/*`] : []), // must exactly match how the revision was encoded in the URL
        '*',
    ]
```

Reply:
Hm I see I think the problem is that we have to prefix the repo name because otherwise paths like `*/-/*` won't match properly, is this correct? 🤔 

One potential workaround that I was thinking was that instead of the splat for the repoAndRevision we do something like this:

```
"/:one/:two?/:three?/:four?/:five?/:six?/:seven?/:eight?/:nine?/:ten?-/*",
```

Reply:
Another problem is that the `rawRevision` rule won't work when it's relative but I’m not sure if we even need it if it matches in the expanded repoAndRevision splat instead?

Anyway I committed something that made the test pass for me haha I think it's incomplete and might break other things though but perhaps you can use something from it 😅 

I tried replacing the root level `/*` splat for the repo container with the above splat and things started to crackle already but maybe I was just missing it? Anyway the thing I commited does pass the test for me:

```
  🖥  Browser console: blob
        ✓ works with spaces in the repository name (3274ms)


    1 passing (4s)
```

Reply:
>Another problem is that the rawRevision rule won't work when it's relative but I’m not sure if we even need it if it matches in the expanded repoAndRevision splat instead?

Could you share an example of the URL you have in mind?

Reply:
It seems I found the root cause of the issue in the react-router codebase. I created a PR with the unit test that reproduces our problem and a potential fix:
- https://github.com/remix-run/react-router/pull/10028

It breaks special-characters tests in react-router, but I decided to share still it to potentially get early thoughts from the RR team on how to approach it.

Reply:
@philipp-spiess, lul, it looks like this super long splat is the best workaround for now. I applied it to other `RepoContainer` routes and pushed it to [this PR](https://github.com/sourcegraph/sourcegraph/pull/47014). Waiting for CI 🤞

Reply:
> Another problem is that the rawRevision rule won't work when it's relative but I’m not sure if we even need it if it matches in the expanded repoAndRevision splat instead?

Oh yeah so when we do relative routes like this:

`<Route path="foobar"/>`

It will always be appended as a new segment so `/we/are/here/foobar`

My thinking with this comment was that if we use `<Route path="@{rawREvision" />` it would match `/my/sg/repo/@revision` and not `/my/sg/repo@revision`.

But later in the comment i realized `the splat will match it anyway and there's probably no need to match it expclicitly.